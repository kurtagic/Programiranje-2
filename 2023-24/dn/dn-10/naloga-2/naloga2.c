
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

#include "naloga2.h"

Tocka projekcija(Tocka t, Premica p) {
    // popravite / dopolnite ...
    return (Tocka) { (Ulomek) {0, 1}, (Ulomek) {0, 1} };
}

#ifndef test

int main() {
    return 0;
}

#endif
