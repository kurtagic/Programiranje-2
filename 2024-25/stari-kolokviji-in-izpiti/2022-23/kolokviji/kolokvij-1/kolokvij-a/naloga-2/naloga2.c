
/*
 * Prevajanje in zagon testnega programa testXY.c:
 *
 * gcc -D=test testXY.c naloga2.c
 * ./a.out
 *
 * Zagon testne skripte ("sele potem, ko ste prepri"cani, da program deluje!):
 *
 * export name=naloga2
 * make test
 *
 * Javni testni primeri (po te"zavnosti):
 * -- 02, 03, 04: en sam vhodni niz;
 * -- 05, 06, 07: vsi vhodni nizi so enako dolgi;
 * -- 01, 08, 09, 10: splo"sni primeri.
 *
 * Javni testni primeri (po na"cinu priprave):
 * -- 01: primer iz besedila;
 * -- 01, 02, 05: ro"cno ustvarjeni;
 * -- ostali: samodejno generirani.
 */

#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

// po potrebi dopolnite ...

//=============================================================================

char** poStolpcih(char** nizi, int stVhodnih, int* stIzhodnih) {
    // popravite / dopolnite ...
    return NULL;
}

//=============================================================================

#ifndef test

int main() {
    // "Ce "zelite funkcijo poStolpcih testirati brez testnih primerov,
    // dopolnite to funkcijo in prevedite datoteko na obi"cajen na"cin
    // (gcc naloga2.c).
    return 0;
}

#endif
