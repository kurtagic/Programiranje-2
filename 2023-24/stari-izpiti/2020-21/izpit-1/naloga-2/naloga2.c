#include <stdio.h>
#include <stdlib.h>

#include "naloga2.h"

Vozlisce *sticisce(Vozlisce *a, Vozlisce *b)
{
    Vozlisce *current_a = a;

    while (current_a != NULL)
    {
        Vozlisce *current_b = b;
        while (current_b != NULL)
        {
            if (current_a == current_b)
            {
                return current_a;
            }

            current_b = current_b->n;
        }

        current_a = current_a->n;
    }

    return NULL;
}

// TESTS
Vozlisce *naredi(int vred)
{
    Vozlisce *novo = malloc(sizeof(Vozlisce));
    novo->vrednost = vred;
    return novo;
}

int main()
{
    Vozlisce *prvo = naredi(1);
    Vozlisce *drugo = naredi(2);
    Vozlisce *tretje = naredi(3);
    Vozlisce *skupno = naredi(10);

    Vozlisce *cetrto = naredi(4);
    Vozlisce *peto = naredi(5);

    prvo->n = drugo;
    drugo->n = tretje;
    tretje->n = skupno;
    cetrto->n = skupno;

    printf("%d\n", sticisce(prvo, cetrto)->vrednost);

    return 0;
}
