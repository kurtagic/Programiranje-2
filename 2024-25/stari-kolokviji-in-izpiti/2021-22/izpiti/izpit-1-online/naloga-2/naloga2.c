
/*
Ro"cno poganjanje testnih programov (npr. test01.c):

gcc -Dtest test01.c naloga2.c
./a.out

Samodejno testiranje:

export name=naloga2
make test

Testni primeri:

test01: primer iz besedila
test02..test08: ro"cno izdelani kratki testi
test09..test10: samodejno izdelani, zadnji element prvega seznama < prvi element drugega seznama
test11..test16: samodejno izdelani, splo"sni

.c: testni program (prebere testne podatke, pokli"ce va"so funkcijo in izpi"se rezultat)
.dat: testni podatki
.out: pri"cakovani izhod
.res: izhod va"sega programa (pri poganjanju z make)
*/

#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

#include "naloga2.h"

// Lahko dodate "se kak #include, pomo"zno funkcijo ipd.

//============================================================================

Vozlisce* zlij(Vozlisce* a, Vozlisce* b) {
    // dopolnite ...

    return NULL;   // popravite
}

//============================================================================

// Vrstici z #ifndef in #endif pustite pri miru!

#ifndef test

int main() {
    return 0;
}

#endif
