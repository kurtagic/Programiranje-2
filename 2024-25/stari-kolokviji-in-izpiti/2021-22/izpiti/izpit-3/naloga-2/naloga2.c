
/*
Ro"cno poganjanje testnih programov (npr. test01.c):

gcc -Dtest test01.c naloga2.c
./a.out

Samodejno testiranje:

export name=naloga2
make test

Testni primeri:

01: primer iz besedila
02..04: ena sama vrstica
05..06: vsaka neprazna vrstica je naslovna
07..10: splo"sni primeri

Datoteke:

test*.c: testni program (prebere testne podatke, pokli"ce va"so funkcijo in izpi"se rezultat)
test*.dat: testni podatki
test*.out: pri"cakovani izhod testnega programa
test*.res: dejanski izhod testnega programa (pri poganjanju z make)
*/

#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

#include "naloga2.h"

// Lahko dodate "se kak #include, pomo"zno funkcijo ipd.

//============================================================================

void vstaviH1(Vozlisce* zacetek) {
    // dopolnite ...
}

//============================================================================

// Vrstici z #ifndef in #endif pustite pri miru!

#ifndef test

int main() {
    // Tole dopolnite, "ce "zelite funkcijo lastnoro"cno preveriti.
    // V tem primeru program po"zenite preprosto kot
    // gcc naloga3.c
    // ./a.out
    return 0;
}

#endif
