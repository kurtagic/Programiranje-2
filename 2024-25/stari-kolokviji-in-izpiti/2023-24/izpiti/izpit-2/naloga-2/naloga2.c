
/*
 * Prevajanje in zagon testnega programa testXY.c:
 *
 * gcc -Dtest testXY.c naloga2.c
 * ./a.out
 *
 * Zagon testne skripte ("sele potem, ko ste prepri"cani, da program deluje!):
 *
 * export name=naloga2
 * make test
 *
 * Javni testni primeri:
 *
 * 02..03: zaporedje podatkov a_1, a_1, a_2, a_2, a_3, a_3, ... (a_i != a_{i+1})
 * 04..05: nikoli nimamo ve"c kot dveh zaporednih vozli"s"c z enako vrednostjo
 *         komponente <podatek>
 * 01, 06..10: splo"sni primeri
 */

#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

#include "naloga2.h"

//=============================================================================

// po potrebi dopolnite ...

void obdelaj(Vozlisce* zacetek) {
    // dopolnite ...
}

#ifndef test

int main() {
    // "Ce "zelite funkcijo <obdelaj> testirati brez testnih primerov,
    // dopolnite to funkcijo in prevedite datoteko na obi"cajen na"cin 
    // (gcc naloga2.c).
    return 0;
}

#endif
