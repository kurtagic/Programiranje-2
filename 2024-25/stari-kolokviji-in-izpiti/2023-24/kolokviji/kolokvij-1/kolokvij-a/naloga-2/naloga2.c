#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

#include "naloga2.h"

void racionaliziraj(char** nizi)
{
	// poglej vse kombinacije parov
	for (int i = 0; nizi[i] != NULL; i++)
	{
		for (int j = 0; nizi[j] != NULL; j++)
		{
			if (i == j)
			{
				continue;
			}

			// ce sta niza enaka, spremen j-ti pointer da kaze na i-ti pointer (vedno j > i)
			if (!strcmp(nizi[i], nizi[j]))
			{
				nizi[j] = nizi[i];
			}
		}
	}
}

#ifndef test
int main() { return 0; }
#endif
