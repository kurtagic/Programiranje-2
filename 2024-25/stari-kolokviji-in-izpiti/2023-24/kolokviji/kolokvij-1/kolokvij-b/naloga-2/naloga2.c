
/*
 * Prevajanje in zagon testnega programa testXY.c:
 *
 * gcc -Dtest testXY.c naloga2.c
 * ./a.out
 *
 * Zagon testne skripte ("sele potem, ko ste prepri"cani, da program deluje!):
 *
 * export name=naloga2
 * make test
 */

#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

#include "naloga2.h"

//=============================================================================

// po potrebi dopolnite ...

void razbohoti(char** nizi) {
    // dopolnite ...
}

//=============================================================================

#ifndef test

int main() {
    // "Ce "zelite funkcijo <razbohoti> testirati brez testnih primerov,
    // dopolnite to funkcijo in prevedite datoteko na obi"cajen na"cin 
    // (gcc naloga2.c).
    return 0;
}

#endif
