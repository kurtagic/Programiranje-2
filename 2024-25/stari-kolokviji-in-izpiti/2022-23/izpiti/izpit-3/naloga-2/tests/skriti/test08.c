
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

#include "naloga2.h"

int* SLIKA[] = {
    (int[]) {462, 771, 1101, 1473, 1645, 1897, 1867, 1931, 1598, 1472, 1044, 692, 594, 273},
    (int[]) {550, 705, 1210, 1498, 1786, 1944, 2023, 1878, 1838, 1434, 1197, 820, 588, 245},
    (int[]) {514, 773, 1048, 1456, 1585, 1844, 2016, 1967, 1712, 1326, 1006, 802, 427, 254},
};

int N = 3;
int M = 14;
int D = 1;

int main() {
    int** nova = zmehcaj(SLIKA, N, M, D);
    int noviN = N - 2 * D;
    int noviM = M - 2 * D;

    for (int i = 0; i < noviN; i++) {
        for (int j = 0; j < noviM; j++) {
            printf("%5d", nova[i][j]);
        }
        printf("\n");
    }

    for (int i = 0; i < noviN; i++) {
        free(nova[i]);
    }
    free(nova);

    return 0;
}
