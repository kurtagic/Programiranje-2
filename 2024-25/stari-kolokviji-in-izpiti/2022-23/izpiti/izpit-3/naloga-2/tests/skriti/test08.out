  792 1115 1422 1680 1845 1929 1870 1684 1403 1088  796  521
